src/utils/SpokeUtils.ts

```diff
@@ -380,3 +384,37 @@ export async function findFillBlock(
 
   return lowBlockNumber;
 }
+
+/*
+ * Determines if the relay data provided contains bytes32 for addresses or standard evm 20-byte addresses.
+ * Returns true if the relay data has bytes32 address representations.
+ */
+export function isUpdatedRelayData(relayData: RelayData) {
+  const isValidBytes32 = (maybeBytes32: string) => {
+    return ethersUtils.isBytes(maybeBytes32) && maybeBytes32.length === 66;
+  };
+  // Return false if the depositor is not a bytes32. Assume that if any field is a bytes32 in relayData, then all fields will be bytes32 representations.
+  return isValidBytes32(relayData.depositor);
+}
+
+/*
+ * Converts an input relay data to to the version with 32-byte address representations.
+ */
+export function convertToUpdatedRelayData(relayData: RelayData): RelayData {
+  return isUpdatedRelayData(relayData)
+    ? relayData
+    : {
+        ...relayData,
+        depositor: toBytes32(relayData.depositor),
+        recipient: toBytes32(relayData.recipient),
+        exclusiveRelayer: toBytes32(relayData.exclusiveRelayer),
+        inputToken: toBytes32(relayData.inputToken),
+        outputToken: toBytes32(relayData.outputToken),
+        message: isMessageEmpty(relayData.message) ? ZERO_BYTES : ethersUtils.keccak256(relayData.message),
+      };
+}
```

Comment:
I'm wondering if we instead want to have everything in its natural representation? If we promote everything to bytes32-formatting (effectively just adding leading 0s to EVM addresses) then our logging will suddenly look awful; long and unreadable.

I'm wondering if it could instead be preferable to normalise all the data on ingestion into the SpokePool and convert EVM addresses as bytes32 back into normal 40-char addresses. The fact that we're working with strings makes this seem more achievable than in the contracts, where we're forced to go to bytes32 for a universal type.

Reply:
The only issue with that is we can't convert from new to old if the new came from a `FilledRelay` event, since then we only have the message hash. We may not need this function, though. I had it for calculating the new relay hash type, which is now not necessary.

Reply:
I removed it since we likely won't need it post contract changes. https://github.com/across-protocol/sdk/pull/830/commits/59582b35d542cf00c4fb60c71c0025b0e5e4e3fd